src/PowerShellEditorServices.Host/EditorServicesHost.cs

```diff
@@ -86,40 +90,74 @@ public class EditorServicesHost
         /// <param name="bundledModulesPath">Provides a path to PowerShell modules bundled with the host, if any.  Null otherwise.</param>
         /// <param name="waitForDebugger">If true, causes the host to wait for the debugger to attach before proceeding.</param>
+        /// <param name="additionalModules">Modules to be loaded when initializing the new runspace.</param>
+        /// <param name="featureFlags">Features to enable for this instance.</param>
         public EditorServicesHost(
             HostDetails hostDetails,
             string bundledModulesPath,
             bool enableConsoleRepl,
             bool waitForDebugger,
             string[] additionalModules,
             string[] featureFlags)
+            : this(
+                hostDetails,
+                bundledModulesPath,
+                enableConsoleRepl,
+                waitForDebugger,
+                additionalModules,
+                featureFlags,
+                GetInternalHostFromDefaultRunspace())
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the EditorServicesHost class and waits for
+        /// the debugger to attach if waitForDebugger is true.
+        /// </summary>
+        /// <param name="hostDetails">The details of the host which is launching PowerShell Editor Services.</param>
+        /// <param name="bundledModulesPath">Provides a path to PowerShell modules bundled with the host, if any.  Null otherwise.</param>
+        /// <param name="waitForDebugger">If true, causes the host to wait for the debugger to attach before proceeding.</param>
+        /// <param name="additionalModules">Modules to be loaded when initializing the new runspace.</param>
+        /// <param name="featureFlags">Features to enable for this instance.</param>
+        /// <param name="internalHost">The value of the $Host variable in the original runspace.</param>
+        public EditorServicesHost(
+            HostDetails hostDetails,
+            string bundledModulesPath,
+            bool enableConsoleRepl,
+            bool waitForDebugger,
+            string[] additionalModules,
+            string[] featureFlags,
+            PSHost internalHost)
         {
             Validate.IsNotNull(nameof(hostDetails), hostDetails);
+            Validate.IsNotNull(nameof(internalHost), internalHost);
 
             this.hostDetails = hostDetails;
             this.enableConsoleRepl = enableConsoleRepl;
             this.bundledModulesPath = bundledModulesPath;
             this.additionalModules = additionalModules ?? new string[0];
             this.featureFlags = new HashSet<string>(featureFlags ?? new string[0]);
             this.serverCompletedTask = new TaskCompletionSource<bool>();
+            this.internalHost = internalHost;
 
 #if DEBUG
             if (waitForDebugger)
             {
-                if (Debugger.IsAttached)
+                if (System.Diagnostics.Debugger.IsAttached)

```

Comment:
do we still need the `using System.Diagnostics` above anymore if you're doing this?

Reply:
Just checked, `Process` and `FileVersionInfo` use it still